### PR TITLE
Fixed bug on len method, implemented getitem method

### PR DESCRIPTION
**Pull request by LTKills**

### CODE REVIEW

File: float_range/range.py [hunk 1]
```diff
@@ -63,11 +64,27 @@ def __ne__(self, other):
...
+        if self._is_empty() or raw_len < 1:
             return 0
 
-        raw_len = abs(self.major - self.minor)
-        return int(raw_len/self.step)
+        return int(math.ceil(raw_len))
+
+    def __getitem__(self, key):
+        if not isinstance(key, int):
+            raise TypeError
```
Comment (nathan-cruz77) [inline float_range/range.py]: Some error message here would be nice

File: float_range/range.py [hunk 2]
```diff
@@ -63,11 +64,27 @@ def __ne__(self, other):
...
+        pos_item = self.step * key + self.start
+        neg_item = self.step * (len(self) + key) + self.start
+
+        if pos_item in self:
+            return pos_item
+
+        if neg_item in self:
+            return neg_item
+
+        raise IndexError
```
Comment (nathan-cruz77) [inline float_range/range.py]: Same thing here. Perhaps we could make something like `range`:
``` python
>>> x = range(10)
>>> x[-1]
9
>>> x[-2]
8
>>> x[10]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: range object index out of range
```
So something like:
```python
class_name = self.__class__.__name__
raise IndexError('{} object index out of range'.format(class_name))
```

File: tests/test_range.py [hunk 3]
```diff
@@ -5,6 +5,18 @@
...
+        expected_value = 1.5
+
+        given_value = FloatRange(0, 10, 1.5)
+        self.assertEqual(expected_value, given_value[1])
+
+    def test_getitem_negative(self):
+        expected_value = 9
+
+        given_value = FloatRange(0, 10, 1.5)
+        self.assertEqual(expected_value, given_value[-1])
```
Comment (nathan-cruz77) [inline tests/test_range.py]: There should be a test covering the case were an exception occurs.

**Comment by LTKills** [comment]
@nathan-cruz77 need to remerge this so the tests pass

**Pull request merged**